### PR TITLE
Add Open Information Extraction

### PR DESCRIPTION
**Pull request by gabrielStanovsky**
Adds Open Information Extraction to AllenNLP, towards adding the trained model to the online demo. 
Follows closely the SRL model:

- Dataset reader - data conforms to SRL, so uses the SRL dataset reader.
- Model - Similar to SRL, no span loss.
- Predictor - Allows multi word predicate spans, conforms to the output format of SRL, in order to reuse its demo components.

### CODE REVIEW

**Comment by joelgrus** [comment]
as discussed offline, this model is so similar to the existing SRL model that we should just use that one (and add a constructor option to disable the irrelevant span metrics) rather than duplicating its code

**Comment by gabrielStanovsky** [comment]
@joelgrus 
I now use the (very slightly modified) SRL model instead. 
Also added a script to preprocess Open IE extractions to CoNLL format. 
PTAL, thanks!

File: allennlp/predictors/open_information_extraction.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                    chunk.append(tag[2:] + ": " + token)
+                elif tag == "O":
+                    frame.append(token)
+
+        if chunk:
+            frame.append("[" + " ".join(chunk) + "]")
+
+        return " ".join(frame)
+
+    def _json_to_instance(self, json_dict: JsonDict) -> Tuple[Instance, JsonDict]:
```
Comment (joelgrus) [inline allennlp/predictors/open_information_extraction.py]: I think this is the wrong return type?

File: allennlp/models/semantic_role_labeler.py [hunk 2]
```diff
@@ -141,7 +145,8 @@ def forward(self,  # type: ignore
                                                       tags,
                                                       mask,
                                                       label_smoothing=self._label_smoothing)
-            self.span_metric(class_probabilities, tags, mask)
+            if not self.ignore_span_metric:
+                self.span_metric(class_probabilities, tags, mask)
```
[the remark this answers is not in the document]
Comment (joelgrus) [reply allennlp/models/semantic_role_labeler.py]: it gets called automatically by the trainer, just have it return an empty dict

**Comment by gabrielStanovsky** [comment]
I just addressed your comments. Thanks!

**Pull request merged**